developerTools/scriptReferences/entityClientScript/entityClientScript.js

```diff
@@ -0,0 +1,116 @@
+// Use as a template for your client entity scripts
+//
+// Client entity scripts are run on a interface instance. Meaning each avatar connected
+// to a domain have one instance of this script running locally and one client controls
+// an entity at one time called "ownership over that entity".
+// Best for physics changes.
+//
+// Copyright 2019 High Fidelity, Inc.
+//
+// Distributed under the Apache License, Version 2.0.
+// See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+(function() {
+    var I_AM_STATIC_VARIABLE = 200;
+    var _this = null;
+    var _entityID = null;
+
+    function EntityClientScript() {
+        _this = this;
+    }
+
+    EntityClientScript.prototype = {
+        // REQUIRED FUNCTIONS
+        preload: function(entityID) {
+            // runs when this script is refreshed or a
+            // client connects to a domain where this entity is present
+            _entityID = entityID;
+        },
+
+        unload: function() {
+            // triggered when avatar leaves the domain where entity is present
+            // clear any intervals
+            // clear any listeners
+            // reset anything else that needs to be
+        },
+
+        // EVENT LISTENERS FOR THE ENTITY
+
+        // Entity Event Listener Notes: 
+        //     For mouse/trigger click on an entity to do an action
+        //         Use: startNearTrigger + startFarTrigger + clickReleaseOnEntity
+
+        // MOUSECLICK METHODS - not triggered by controller
+        // params are: entityID, event PointerEvent (https://apidocs.highfidelity.com/global.html#PointerEvent)
+        // https://apidocs.highfidelity.com/Entities.html#.clickDownOnEntity
+        clickReleaseOnEntity: function(entityID, pointerEvent) {
+            if (pointerEvent.isPrimaryButton) {} // will only work on left mouse click release event (see PointerEvent)
+        },
+        clickDownOnEntity: function(entityID, pointerEvent) {},
+        holdingClickOnEntity: function(entityID, pointerEvent) {},
+        
+        mouseMoveOnEntity: function(entityID, pointerEvent) {},
+        mousePressOnEntity: function(entityID, event) {},
+        mouseReleaseOnEntity: function(entityID, event) {},
+        mouseDoublePressOffEntity: function(pointerEvent) {},
+
+        hoverEnterEntity: function(entityID, pointerEvent) {},
+        hoverLeaveEntity: function(entityID, pointerEvent) {},
+        hoverOverEntity: function(entityID, pointerEvent) {},
+
+        // triggered when a script in a Web entity's web page script sends and event over hte script's eventbridge
+        webEventReceived: function(entityID, message) {}, // https://apidocs.highfidelity.com/Entities.html#.webEventReceived
+
+        // COLLISION METHODS
+        // Avatar collision with this entity
+        leaveEntity: function(userID) {},
+        enterEntity: function(userID) {}, // https://apidocs.highfidelity.com/Entities.html#.enterEntity
```

Comment:
could you please add links to the docs pages for all of these signals instead of only some? i think that'd be really helpful.